### PR TITLE
feat(SageMaker): Add SageMaker instrumentation

### LINKED ISSUE #1660: 🚀 Feature: Tracing support for Amazon SageMaker endpoints
### Which component is this feature for?

All Packages

### 🔖 Feature description

Add tracing support for [Amazon SageMaker endpoints](https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints.html)

### 🎤 Why is this feature needed ?

Developers may deploy a variety of LLMs as endpoints on Amazon SageMaker. They'd query these endpoints using the [InvokeEndpoint](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_runtime_InvokeEndpoint.html) API to retrieve the LLM response.

### ✌️ How do you aim to achieve this?

1. Create `opentelemetry-instrumentation-sagemaker` package
2. Create instrumentation for the `sagemaker-runtime` service and patch the `invoke_endpoint` method.
3. Create span for the request.

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by bobbywlindsey**
- [x] I have added tests that cover my changes.
- [x] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.
sagemaker-instrumentation-cloudwatch
- [x] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [x] (If applicable) I have updated the documentation accordingly.

Summary of changes (closes #1660):

- Added new instrumentation for Amazon SageMaker endpoints
- Added 2 new semantic conventions:
    1. `llm.sagemaker.request` which captures the request body of the endpoint invocation
    2. `llm.sagemaker.response` which captures the response body of the endpoint invocation

### CODE REVIEW

**Review by nirga — COMMENTED** [review]
Thanks so much @bobbywlindsey! Looking good :)
Left a couple of small comments

File: packages/opentelemetry-instrumentation-sagemaker/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# OpenTelemetry Bedrock Instrumentation
+
+<a href="https://pypi.org/project/opentelemetry-instrumentation-bedrock/">
```
[the remark this answers is not in the document]
Comment (bobbywlindsey) [reply packages/opentelemetry-instrumentation-sagemaker/README.md:3]: Ah totally missed this file, thanks for the catch. Will update.

File: packages/opentelemetry-instrumentation-sagemaker/opentelemetry/instrumentation/sagemaker/__init__.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import os
+from typing import Collection
+from opentelemetry.instrumentation.sagemaker.config import Config
+from opentelemetry.instrumentation.sagemaker.reusable_streaming_body import (
+    ReusableStreamingBody,
+)
+from opentelemetry.instrumentation.sagemaker.streaming_wrapper import StreamingWrapper
+from opentelemetry.instrumentation.sagemaker.utils import dont_throw
+from wrapt import wrap_function_wrapper
+import anthropic
```
[the remark this answers is not in the document]
Comment (bobbywlindsey) [reply packages/opentelemetry-instrumentation-sagemaker/opentelemetry/instrumentation/sagemaker/__init__.py:15]: Ah nope, I'll remove it 👍🏻

File: packages/opentelemetry-instrumentation-sagemaker/opentelemetry/instrumentation/sagemaker/__init__.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        response["Body"]._raw_stream, response["Body"]._content_length
+    )
+    request_body = json.loads(kwargs.get("Body"))
+    response_body = json.loads(response.get("Body").read())
+
+    endpoint_name = kwargs.get("EndpointName")
+
+    _set_span_attribute(span, SpanAttributes.LLM_REQUEST_MODEL, endpoint_name)
+    _set_span_attribute(
+        span, SpanAttributes.LLM_SAGEMAKER_REQUEST, json.dumps(request_body)
```
[the remark this answers is not in the document]
Comment (bobbywlindsey) [reply packages/opentelemetry-instrumentation-sagemaker/opentelemetry/instrumentation/sagemaker/__init__.py:164]: SGTM, will make the updates.

**Comment by nirga** [comment]
@bobbywlindsey seems like tests are failing - I think you'll need to amend the cassettes or the mock API URL you're using

**Comment by bobbywlindsey** [comment]
@nirga Thanks for the help :) I think I've resolved the build issues now. Let me know if there's anything else I can do to improve the PR. Once this one is merged, I'll begin working on Bedrock Knowledge Bases and Agents issues 👍🏻

**Review by nirga — APPROVED** [review]

**Pull request merged**